### DISCUSSION TITLE
Warning: Cannot update a component (`Controller`) while rendering a different component (`EditEventPage`).

### DISCUSSION THREAD

**Discussion by Ekeu** [react-hook-form/react-hook-form#5007] [Q&A]
Hello react-hook-form community 🙋🏽‍♂️

I have a form which I use to register events. While learning react-hook-form, I read about the `Controller` component that helps integrate other ui libraries like react-select easily.

Everything worked fine while doing the registration feature. This is how it looked like : 

![example-3](https://user-images.githubusercontent.com/23312714/116664769-32a9a080-a999-11eb-804c-f19e69c09bd1.png)

The `GooglePlacesAutocomplete` actually uses `react-select`

When I tried implementing the ability to update an event, meaning using the previous form but in a different component; at the beginning of my component; I did this:

![example-4](https://user-images.githubusercontent.com/23312714/116665227-cd09e400-a999-11eb-9899-3808ec3478a2.png)

I totally agree, you guys clearly said in the docs it is recommended to register before setting any value even though it is permitted.

![image](https://user-images.githubusercontent.com/23312714/116665601-4275b480-a99a-11eb-8848-ab7175eea494.png)

Even though it works, I actually get this warning:

![image](https://user-images.githubusercontent.com/23312714/116665775-751fad00-a99a-11eb-88cd-910181b00ee1.png)

I not really sure about the reason I getting this, but I think it's the `setValue('address', JSON.parse(event.addressObject));` that updates the `Controller` component at the same time my component tries to render.

If I'm right then how can I register the Controller before? Can I do something like this ==> `const { onChange, ...rest } = register('address'); `

Thank you!

**Comment by bluebill1049** [comment]
can you share a codesandbox? We can take a look at it.
